### ISSUE TITLE
`DuplicateBasesError: Duplicates found in MROs` when inferring enum class with duplicate ancestors

### ISSUE THREAD

**Issue by correctmost** [pylint-dev/astroid#3065]
### Steps to reproduce

The following code triggers a `DuplicateBasesError` exception that is not caught by Pylint:

```python
import enum

class C(enum.Enum, enum.Enum):
    pass
```

I found the bug during a local fuzzing run.

### Current behavior
```python-traceback
Traceback (most recent call last):
  File "pylint/pylint/lint/pylinter.py", line 1039, in get_ast
    return MANAGER.ast_from_file(filepath, modname, source=True)
           ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "astroid/astroid/manager.py", line 165, in ast_from_file
    return AstroidBuilder(self).file_build(filepath, modname)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^
  File "astroid/astroid/builder.py", line 149, in file_build
    return self._post_build(module, builder, encoding)
           ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "astroid/astroid/builder.py", line 177, in _post_build
    module = self._manager.visit_transforms(module)
  File "astroid/astroid/manager.py", line 129, in visit_transforms
    return self._transform.visit(node)
           ~~~~~~~~~~~~~~~~~~~~~^^^^^^
  File "astroid/astroid/transforms.py", line 163, in visit
    return self._visit(node)
           ~~~~~~~~~~~^^^^^^
  File "astroid/astroid/transforms.py", line 85, in _visit
    visited = self._visit_generic(value)
  File "astroid/astroid/transforms.py", line 113, in _visit_generic
    return [self._visit_generic(child) for child in node]
            ~~~~~~~~~~~~~~~~~~~^^^^^^^
  File "astroid/astroid/transforms.py", line 120, in _visit_generic
    return self._visit(node)
           ~~~~~~~~~~~^^^^^^
  File "astroid/astroid/transforms.py", line 88, in _visit
    return self._transform(node)
           ~~~~~~~~~~~~~~~^^^^^^
  File "astroid/astroid/transforms.py", line 68, in _transform
    ret = transform_func(node)
  File "astroid/astroid/brain/brain_namedtuple_enum.py", line 389, in infer_enum_class
    for basename in (b for cls in node.mro() for b in cls.basenames):
                                  ~~~~~~~~^^
  File "astroid/astroid/nodes/scoped_nodes/scoped_nodes.py", line 2918, in mro
    return self._compute_mro(context=context)
           ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "astroid/astroid/nodes/scoped_nodes/scoped_nodes.py", line 2906, in _compute_mro
    unmerged_mro = clean_duplicates_mro(unmerged_mro, self, context)
  File "astroid/astroid/nodes/scoped_nodes/scoped_nodes.py", line 156, in clean_duplicates_mro
    raise DuplicateBasesError(
    ...<4 lines>...
    )
astroid.exceptions.DuplicateBasesError: Duplicates found in MROs (C), (Enum, object), (Enum, object), (Enum, Enum) for <ClassDef.C l.3 at 0x7f410a6a0500>.
```

### Expected behavior
No crash

### Version
649ec7d37d182
